### PR TITLE
fix: ensure existence of urn when calling BaseEntityResource::get

### PR DESCRIPTION
**Pull request by jsdonn**
motivation: calling get() with an empty, non-null aspectNames parameter (i.e. []) results in a 404, even if the entity exists. the issue stems from the call to getInternalNonEmpty which filters out urn/aspect mappings where the value is empty (which it will be when aspectNames is empty). this results in calling get(0) on an empty map, which results in a 404.

the intended behavior should be: 
urn doesn't exist - 404
urn exists -
    * null aspects param - return all aspects
    * empty aspects param - return base response with no aspects (i.e. only non-aspect fields)
this change checks that the urn exists in the db before calling one of the getInternal methods. additionally, because we do this exists check, we can now use getInternal instead of getInternalNonEmpty to get our intended behavior.

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java [hunk 1]
```diff
@@ -147,7 +147,10 @@ protected BaseRestliAuditor getAuditor() {
 
     return RestliUtils.toTask(() -> {
       final URN urn = toUrn(id);
-      final VALUE value = getInternalNonEmpty(Collections.singleton(urn), parseAspectsParam(aspectNames)).get(urn);
+      if (!getLocalDAO().exists(urn)) {
+        throw RestliUtils.resourceNotFoundException();
+      }
+      final VALUE value = getInternal(Collections.singleton(urn), parseAspectsParam(aspectNames)).get(urn);
```
[the remark this answers is not in the document]
Comment (jsdonn) [reply restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java:153]: Thanks for the comments, I will go through with the change then. I will add some test cases and update when I've done so.

File: restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java [hunk 2]
```diff
@@ -147,7 +147,10 @@ protected BaseRestliAuditor getAuditor() {
 
     return RestliUtils.toTask(() -> {
       final URN urn = toUrn(id);
-      final VALUE value = getInternalNonEmpty(Collections.singleton(urn), parseAspectsParam(aspectNames)).get(urn);
```
[the remark this answers is not in the document]
Comment (jsdonn) [reply restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java:150]: I'm going to say no for now. I see that `getInternalNonEmpty` is being called in `getAll` and `search`. I've already created a ticket in backlog to keep track of needing consistency between getAll, batch get, and regular get methods. I'll add this to the list.

File: restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsdonn) [reply restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java:153]: Test cases added. Please review when you get a chance, thanks.

**Review by RealChrisL — APPROVED** [review]

File: restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jywadhwani) [reply restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java:150]: Sounds good.

File: restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java [hunk 3]
```diff
@@ -163,6 +166,7 @@ protected BaseRestliAuditor getAuditor() {
   public Task<Map<KEY, VALUE>> batchGet(
       @Nonnull Set<KEY> ids,
       @QueryParam(PARAM_ASPECTS) @Optional @Nullable String[] aspectNames) {
+    // TODO: META-15492 - discuss and sync with get()'s intended behavior (check if urn exists).
```
Comment (jywadhwani) [inline restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java:169]: let us not refer to (internal) META tickets here. You could create an issue [here](https://github.com/linkedin/datahub-gma/issues) and refer to that issue. Internally you can create a META ticket and refer to the issue you created here.

Comment (jsdonn) [reply restli-resources/src/main/java/com/linkedin/metadata/restli/BaseEntityResource.java]: My bad. I overlooked this. Fixed now.

**Review by jywadhwani — APPROVED** [review]

**Pull request merged**